### PR TITLE
Developer doc

### PR DESCRIPTION
**Pull request by julien-nc**
* add developer doc in `/docs`
* write and generate OpenAPI specs
* make sure frontend helper functions are not broken, call new ones in the old ones
* harmonize STT scheduling logic with the rest (properly schedule, factorize frontend code, return task in the API etc...)

### CODE REVIEW

File: lib/Controller/AssistantApiController.php [hunk 1]
```diff
@@ -108,9 +126,14 @@ public function getAssistantTask(int $metaTaskId): DataResponse {
 	/**
 	 * Get user's tasks
 	 *
-	 * @param string|null $taskType
-	 * @param int|null $category
+	 * Get a list of assistant tasks for the current user.
+	 *
+	 * @param string|null $taskType Task type id. If null, tasks of all task types will be retrieved (default: null)
+	 * @param int|null $category Task category. If null, tasks of all categories will be retrieved (default: null)
```
Comment (provokateurin) [inline lib/Controller/AssistantApiController.php:132]:
```suggestion
	 * @param string|null $taskType Task type id. If null, tasks of all task types will be retrieved
	 * @param int|null $category Task category. If null, tasks of all categories will be retrieved
```
The default value will already be in the spec and displayed correctly

File: lib/Controller/Text2ImageApiController.php [hunk 2]
```diff
@@ -42,14 +42,17 @@ public function __construct(
...
-	 * @param bool $displayPrompt
-	 * @param bool $notifyReadyIfScheduled
-	 * @param bool $schedule
+	 * @param string $appId App id to be set in the created task
+	 * @param string $identifier Identifier to be set in the created task
+	 * @param string $prompt Input prompt for the image
+	 * @param int $nResults Number of images to generate (default: 1)
+	 * @param bool $displayPrompt Option to include the prompt when displaying the result images (default: false)
+	 * @param bool $notifyReadyIfScheduled Whether a notification will be produced when the tasks has run if it was scheduled (default: false)
+	 * @param bool $schedule Force scheduling even if the task could run synchronously (default: false)
```
Comment (provokateurin) [inline lib/Controller/Text2ImageApiController.php:51]: Same here, default values don't need to be in the doc comment.

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: docs/web-integration.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+}
+```
+
+Complete example:
+``` javascript
+OCA.Assistant.openAssistantForm({
+	appId: 'my_app_id',
+	identifier: 'my custom identifier',
+	taskType: 'OCP\\TextProcessing\\FreePromptTaskType',
+	input: 'count to 3',
```
Comment (kyteinsky) [inline docs/web-integration.md:68]: can be changed to `inputs` ig. Will allow for much more flexibility. But maybe not something for this PR.

Comment (julien-nc) [reply docs/web-integration.md:68]: Yes it will be done soon, thx.

File: docs/web-integration.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+## Displaying a task result in the assistant
+
+There are 2 ways to display a task with the assistant.
+
+### Open the assistant modal
+
+If you get a task with the `/ocs/v2.php/apps/assistant/api/v1/task/TASK_ID` or `/ocs/v2.php/apps/assistant/api/v1/tasks` OCS endpoint,
+you can display it in the assistant UI by using the `OCA.Assistant.openAssistantTask` helper function.
+This function takes the task object as parameter. Calling it will open the assistant and load the task values
```
Comment (kyteinsky) [inline docs/web-integration.md:16]: Maybe we can have a link here for the task object to point to the below section? And mention it explicitly there?

**Review by kyteinsky — APPROVED** [review]
lgtm!

**Pull request merged**